MoreLinq.Test/EquiZipTest.cs

```diff
@@ -73,43 +73,53 @@ public void ZipWithEqualLengthSequencesFailStrategy()
         }
 
         [Test]
-        [ExpectedException(typeof(InvalidOperationException))]
         public void ZipWithFirstSequenceShorterThanSecondFailStrategy()
         {
-            var zipped = new[] { 1, 2 }.EquiZip(new[] { 4, 5, 6 }, Tuple);
-            Assert.That(zipped, Is.Not.Null);
-            zipped.Consume();
+            Assert.Throws<InvalidOperationException>(() =>
+            {
+                var zipped = new[] { 1, 2 }.EquiZip(new[] { 4, 5, 6 }, Tuple);
+                Assert.That(zipped, Is.Not.Null);
+                zipped.Consume();
```

Comment:
Only this line should be inside the lambda to `Assert.Throws` as that's what will cause and being tested for an exception.